apps/rush-lib/src/cli/logic/pnpm/PnpmLinkManager.ts

```diff
@@ -31,15 +31,19 @@ export class PnpmLinkManager extends BaseLinkManager {
     try {
       const rushLinkJson: IRushLinkJson = { localLinks: {} };
 
+      let promise: Promise<void> = Promise.resolve();
+
       for (const rushProject of this._rushConfiguration.projects) {
-        console.log(os.EOL + 'LINKING: ' + rushProject.packageName);
-        this._linkProject(rushProject, rushLinkJson);
+        promise = promise.then(() => {
+          console.log(os.EOL + 'LINKING: ' + rushProject.packageName);
+          this._linkProject(rushProject, rushLinkJson);
```

Comment:
>this._linkProject(rushProject, rushLinkJson); [](start = 10, length = 45)

don't you need to return this value?

Reply:
Good catch

---
In reply to: [179008412](https://github.com/Microsoft/web-build-tools/pull/596#discussion_r179008412) [](ancestors = 179008412)